accepted/2021/net6.0-tfms/net6.0-tfms.md

```diff
@@ -0,0 +1,396 @@
+
+### Compatibility rules
+
+The [built-in compatibility rules][net5.0] of the `netX.Y-platform` TFMs make a
+TFM automatically compatible with:
+
+1. Itself
+2. Earlier versions of itself
+3. The corresponding `netX.Y` TFM
+4. Earlier versions of the corresponding `netX.Y` TFM
+
+The `net5.0` TFM has the additional twist that it's compatible with
+`netcoreapp3.1` (and earlier), `netstandard2.1` (and earlier) as well as .NET
+Framework (`net4x` and earlier, but under a warning). The .NET Framework
+compatibility is achieved as a *fallback*, that is, it's only used when nothing
+else applies.
+
+We don't want to model the Xamarin TFMs as a fallback but instead want them to
+apply **before** `netcoreapp` and `netstandard`. Usually the existence of the
+Xamarin TFM implementation is to bait a `netstandard` API and switch the
+platform specific implementation. Most of the Plugin type packages use this
+pattern and in fact the `netstandard` "implementation" is more like a reference
+assembly with just stubs that throw at runtime. `Xamarin.Essentials` is a good
+example of this. The same rational can be applied to `netcoreapp` where this
+would logically be the server implementation (or maybe a .NET Core 3.x desktop
+implementation).
+
+A `net6.0-maccatalyst` project referencing a NuGet package should behave as
+follows:
+
+* It should prefer `xamarin.ios` assets over `netcoreapp`, `netstandard`,
+  and `net5.0` assets.
+    - However, it should still prefer `net6.0` assets over `xamarin.ios`.
+    - It also shouldn't be compatible with any other existing Xamarin TFM (such
+      as `xamarin.mac`)
+* Generate NuGet warning [NU1701] when a `xamarin.ios` asset is being used
+    - Package 'packageId' was restored using 'xamarin.ios' instead the project
+      target framework 'net6.0-maccatalyst'. This package may not be fully
+      compatible with your project.
+* Should only use compatible `net5.0` based TFMs, namely `net5.0`, `net6.0`, and
+  `net6.0-maccatalyst`.
+    - Specifically, it should not accept `net6.0-ios`. The expectation is that
+      moving forward libraries that want to work on iOS and Mac Catalyst should
+      use `net6.0` or multi-target for `net6.0-ios` and `net6.0-maccatalyst`
+
+We want the same rules for `net6.0-ios` except that using `xamarin.ios` should
+not generate a warning. This reasoning results in these precedence rules:
+
+**net6.0-ios**
+
+1. `net6.0-ios`
+1. `net6.0`
+1. `xamarin.ios` (no warning)
+1. `net5.0`
+1. `netcoreapp3.1` – `1.0`
+1. `netstandard2.1` – `1.0`
+1. `net4.x` – `1.0` (warning)
+
+**net6.0-maccatalyst**
+
+1. `net6.0-ios`
+1. `net6.0`
+1. `xamarin.ios` ([NU1701] warning)
+1. `net5.0`
+1. `netcoreapp3.1` – `1.0`
+1. `netstandard2.1` – `1.0`
+1. `net4.x` – `1.0 ([NU1701] warning)
+
+**net6.0-macos**
+
+1. `net6.0-macos`
+1. `net6.0` - `5.0`
+1. `netcoreapp3.1` – `1.0`
+1. `netstandard2.1` – `1.0`
+1. `net4.x` – `1.0` ([NU1701] warning)
+
+**net6.0-tvos**
```

Comment:
Why doesn't this support `xamarintvos` assets?

Reply:
A good point, will add.